### PR TITLE
Smarter factory placement for Nation AI 🤖

### PR DESCRIPTION
**Pull request by deshack**
## Description:

Introduces a dedicated `factoryValue()` scoring function for AI factory placement, replacing the generic `interiorStructureValue()` previously shared with cities and missile silos.

Scoring criteria:
- High elevation and spacing from other factories (unchanged from city/silo logic)
- Rail connectivity: bonus per distinct rail cluster reachable within `trainStationMaxRange`, weighted by trade gold potential — allied clusters score highest (1.0), team/neutral clusters score ~0.71, own clusters ~0.29 (based on `config.tradeGold()` values). Based on difficulty
- Cluster deduplication: connecting to the same cluster multiple times does not inflate the score
- Embargoed and bot neighbors are excluded; all other non-embargoed neighbors are included

The result is that the AI tends to place factories where they can bridge separate rail networks or connect to high-value trade partners, rather than deep in its own interior.

### EDIT

Added a dedicated `cityValue()` scoring function that takes into account the connectivity score. This allows placement of cities in a "factory-aware" way, while also enforcing spreading structures (we want the network to grow, not a cluster of cities and factories all together).

## Please complete the following:

- [x] I have added screenshots for all UI updates
- [x] I process any text displayed to the user through translateText() and I've added it to the en.json file
- [x] I have added relevant tests to the test directory
- [x] I confirm I have thoroughly tested these changes and take full responsibility for any bugs introduced

## Please put your Discord username so you can be contacted if a bug or regression is found:

deshack_82603

### CODE REVIEW

**Comment by deshack** [comment]
@coderabbitai review

File: src/core/execution/nation/NationStructureBehavior.ts [hunk 1]
```diff
@@ -559,6 +561,129 @@ export class NationStructureBehavior {
...
+    // Neighbor structures — all non-embargoed non-bot neighbors.
+    for (const neighbor of player.neighbors()) {
+      if (!neighbor.isPlayer()) continue;
+      if (neighbor.type() === PlayerType.Bot) continue;
+      if (!player.canTrade(neighbor)) continue;
+      const relType = player.isOnSameTeam(neighbor)
+        ? "team"
+        : player.isAlliedWith(neighbor)
+          ? "ally"
+          : "other";
```
[the remark this answers is not in the document]
Comment (deshack) [reply src/core/execution/nation/NationStructureBehavior.ts:732]: The order is correct. If you are on the same team, you get the lower gold amount. Therefore the check is consistent with game logic.

**Comment by deshack** [comment]
@coderabbitai review

**Comment by deshack** [comment]
@coderabbitai resume

**Comment by deshack** [comment]
@FloPinguin I needed to tweak the cities value function too, in order to allow a factory-aware placement.

**Comment by deshack** [comment]
@coderabbitai resume

File: src/core/execution/nation/NationStructureBehavior.ts [hunk 2]
```diff
@@ -559,6 +569,287 @@ export class NationStructureBehavior {
...
+            tile: unit.tile(),
+            cluster: unitToCluster.get(unit)!,
+            weight,
+          });
+        }
+      }
+    }
+
+    return result;
+  }
```
[the remark this answers is not in the document]
Comment (deshack) [reply src/core/execution/nation/NationStructureBehavior.ts:750]: All requested tests exist, are correct against the current source, and cover boundary values, inclusion/exclusion rules, weight formulas, and cache semantics.

**Comment by deshack** [comment]
@coderabbitai review
